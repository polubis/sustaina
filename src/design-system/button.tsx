import React from 'react';
import c from 'classnames';

interface ButtonProps
  extends React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > {}

const Button: React.FC<ButtonProps> = ({ className, ...props }) => {
  const classes = c(
    className,
    `bg-gray-300 dark:bg-red-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center`,
  );
  return <button className={classes} {...props} />;
};

export { Button };
