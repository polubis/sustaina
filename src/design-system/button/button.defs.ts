import type React from 'react';

interface ButtonProps
  extends React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > {
  i: 1 | 2 | 3;
  wfull?: boolean;
  rfull?: boolean;
}

export type { ButtonProps };
