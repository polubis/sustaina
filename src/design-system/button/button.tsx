import React from 'react';
import type { ButtonProps } from './button.defs';
import c from 'classnames';

const classes = `px-4 py-2 rounded enabled:focus:outline outline-4 outline-black font-medium font-sans text-medium disabled:cursor-not-allowed`;

const Button: React.FC<ButtonProps> = ({
  className,
  wfull,
  i,
  rfull,
  ...props
}) => {
  const combinedClasses = c(
    className,
    classes,
    { 'w-full': wfull },
    { 'rounded-full': rfull },
  );

  return (
    <button
      className={c(
        combinedClasses,
        {
          [`disabled:bg-orange-600/30 bg-orange-600 text-white enabled:hover:bg-orange-700`]:
            i === 1,
        },
        {
          [`disabled:bg-slate-200 bg-gray-300 text-black enabled:hover:bg-gray-400/70`]:
            i === 2,
        },
        { [`hover:bg-gray-200 text-black`]: i === 3 },
      )}
      {...props}
    />
  );
};

export { Button };
