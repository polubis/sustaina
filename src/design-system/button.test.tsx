import React from 'react';
import { Button } from './button';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe(`Button component works when: `, () => {
  it(`displays content`, () => {
    const { asFragment } = render(
      <Button className="my-class">Content</Button>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it(`allows to pass native button properties and events`, async () => {
    const spy = jest.fn();

    const { asFragment } = render(
      <Button className="my-class" aria-checked="true" onClick={spy}>
        Content
      </Button>,
    );

    await userEvent.click(screen.getByText(/Content/));

    expect(spy).toHaveBeenCalledTimes(1);
    expect(asFragment()).toMatchSnapshot();
  });
});
