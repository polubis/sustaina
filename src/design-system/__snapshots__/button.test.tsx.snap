// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component works when:  allows to pass native button properties and events 1`] = `
<DocumentFragment>
  <button
    aria-checked="true"
    class="my-class bg-gray-300 dark:bg-red-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center"
  >
    Content
  </button>
</DocumentFragment>
`;

exports[`Button component works when:  displays content 1`] = `
<DocumentFragment>
  <button
    class="my-class bg-gray-300 dark:bg-red-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center"
  >
    Content
  </button>
</DocumentFragment>
`;
